plugins/cf_axi_tdd.c

```diff
@@ -123,70 +179,93 @@ static GtkWidget *cf_axi_tdd_init(struct osc_plugin *plugin, GtkWidget *notebook
 	if (!dev)
 		goto context_destroy;
 
-	if (osc_load_glade_file(priv->builder, "cf_axi_tdd") < 0)
-		goto context_destroy;
+	/*check for version device attribute , this only exists in v2 of the tdd driver
+	 * if the attribute is not found , load the previous version of the tdd plugin */
+
+	if(!iio_device_find_attr(dev, "version")) {
+	    if (osc_load_glade_file(priv->builder, "cf_axi_tdd") < 0)
+	      goto context_destroy;
```

Comment:
The indentation looks wrong... Please don't mix styles in the same file/plugin. This one is likely (as I did it :)) using tabs (8 spaces) so please use the same style. It also looks like some widgets have the same attrs (like burst_count and frame_length_ms) so make sure to not duplicate code if not needed.

Also, since for v2 you're initializing all the widgets in cf_axi_tdd_v2_widgets_init(), I would consider in having a precursor patch adding a similar function to the legacy plugin so things are consistent.